sphinx_gallery/backreferences.py

```diff
@@ -121,12 +125,20 @@ def get_mapping(self):
                             for depth in range(len(module), 0, -1):
                                 full_name = '.'.join(
                                     module[:depth] + [class_name] + method)
-                                yield name, full_name, class_attr, is_class
-                elif local_name in self.imported_names:
+                                options.append(
+                                    (name, full_name, class_attr, is_class))
+            # second pass: by import (can't resolve as well without doing
+            # some actions like actually importing the modules, so use it
+            # as a last resort)
+            for split_level in range(len(local_name_split)):
+                local_name = '.'.join(local_name_split[:split_level + 1])
+                remainder = name[len(local_name):]
+                if local_name in self.imported_names:
                     full_name = self.imported_names[local_name] + remainder
-                    is_class = False
-                    yield name, full_name, class_attr, is_class
-                    break
+                    is_class = class_attr = False  # can't tell without import
+                    options.append(
+                        (name, full_name, class_attr, is_class))
+        return options
```

Comment:
@lucyleeow I simplified this a great deal. The diff is a bit ugly, but if you look at the actual file now hopefully it reads much more easily than it did before:

https://github.com/sphinx-gallery/sphinx-gallery/blob/58df0e6d472fab7e6745db6b01b9ee24333b3ab3/sphinx_gallery/backreferences.py#L80-L140